content_scripts/doctolib/book.js

```diff
@@ -23,8 +23,8 @@
     decembre: 12,
   };
 
-  async function waitTimeout(timeout) {
-    await new Promise((r) => setTimeout(r, timeout));
+  function waitTimeout(timeout) {
+    return new Promise((r) => setTimeout(r, timeout));
```

Comment:
Es-tu sûr pour ça ?

Reply:
oui le async assure juste qu'on force le retour d'une promise, il wrap le return dans une promise. Donc pas besoin si on return explicitement une Promise

"The word “async” before a function means one simple thing: a function always returns a promise. Other values are wrapped in a resolved promise automatically."

Reply:
Les deux fonctionnent en effet :)

Moi j'aime bien avoir le `async` explicitement car c'est très clair en regardant la fonction qu'on va obtenir une promise. (c'est pour ça que j'avais écrit comme ça d'ailleurs, mais aussi un peu par habitude). En l'occurrence les 2 écritures sont à la fois équivalentes et identiques en terme de lecture vu que la fonction est petite et qu'il est évident qu'elle retourne une Promise.

Reply:
Je laisserai comme tu avais fait @julienw, ça me semble plus explicite pour les raisons que tu as indiqué.

Reply:
C'est revert 👍